x-pack/plugins/rule_registry/server/alert_data_client/alerts_client.ts

```diff
@@ -345,15 +357,28 @@ export class AlertsClient {
         },
         operation
       );
-      return buildEsQuery(
+      let esQuery;
+      if (id != null) {
+        esQuery = { query: `_id:${id}`, language: 'kuery' };
+      } else if (typeof query === 'string') {
+        esQuery = { query, language: 'kuery' };
+      } else if (query != null && typeof query === 'object') {
+        esQuery = [];
+      }
+      const builtQuery = buildEsQuery(
         undefined,
-        { query: query == null ? `_id:${id}` : query, language: 'kuery' },
+        esQuery == null ? { query: ``, language: 'kuery' } : esQuery,
         [
           (authzFilter as unknown) as Filter,
           ({ term: { [SPACE_IDS]: alertSpaceId } } as unknown) as Filter,
         ],
         config
       );
+      if (query != null && typeof query === 'object') {
+        // @ts-expect-error
+        builtQuery.bool.must.push(query);
```

Comment:
nit: I would avoid a mutation with things like a `push` and use splats or other techniques. If you throw this into another function (for example) that is async, you could mutate it as a side effect when the async call is waiting for a network response as an example.

Reply:
Copied this from here:

https://github.com/elastic/kibana/blob/2d385b339dd334b05eca32c907a2072cc32f8b02/src/plugins/data/common/search/expressions/esdsl.ts#L93-L95